### DISCUSSION TITLE
Dynamic Dropdown

### DISCUSSION THREAD

**Discussion by mcrexy** [eclipse-birt/birt#1679] [Q&A] [answered]
Hi,

How do I display the dropdown? I am able to preview the list when trying to select a default value
![image](https://github.com/eclipse-birt/birt/assets/127509341/de436b23-3b52-4876-8f76-5edf8e599e7d)

But when I go to web viewer, I unable to see any list being dropped.

![image](https://github.com/eclipse-birt/birt/assets/127509341/75ef05af-8c59-4e8a-b934-215a8ddf6d8e)

The dataset contains two columns: ID and Name. I want the user to select the Name, then the related ID is passed as the parameter instead.

Thank you guys in advance!

**Comment by speckyspooky** [comment]
What are the names of the columns because you have written that the statement has 2 columns ID & Name.
But you have added as value and display text the same column.

**Reply by mcrexy** [reply]
Hi @speckyspooky! Thank you for replying.

The two columns are mBOM_ID and Name.

![image](https://github.com/eclipse-birt/birt/assets/127509341/571721d2-1d92-4ae1-a348-27cc31b25979)

But the dropdown doesn't show up even if I put different, what is the step I missed? Is there any guide I can find on how to do dynamic dropdown? I couldn't find it.

**Reply by speckyspooky** [reply, marked as answer]
Attached you will find a small example for the listbox-parameter with a dynamic dataset based on classic-cars from BIRT.
The definition looks very familiar to your definition. May be this will help you to figure out the problem.

[dynamic_parameter.zip](https://github.com/eclipse-birt/birt/files/15339771/dynamic_parameter.zip)

Definition of parameter
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/c4efe936-2990-496c-b33b-851ef2a8ff17)

Preview results
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/fa62c902-4f69-42b9-9791-7aeeae434a99)

**Reply by mcrexy** [reply]
Appreciate it a lot, will get it tested soon!

**Reply by mcrexy** [reply]
Hello sir,

Just got back from overseas, pardon me for the late response. 

I realised that the issue with my rptdesign is that the dataset needs to query from a service. This could be the reason why it always show blank as it's not queried yet. 

![image](https://github.com/eclipse-birt/birt/assets/127509341/e9edb722-1e2b-4ab7-8ee1-8d65a30564f5)

**Reply by speckyspooky** [reply]
Let me know if you have further questions then we can take a look into the open points.

**Reply by mcrexy** [reply]
Hello! Currently I don't have  any question. Thank you for the great help!
